tests/test_operation.py

```diff
@@ -1845,3 +1846,48 @@ def test_eigvals_tensor_deprecation(self):
             m1 = op.eigvals
 
         assert np.allclose(m1, op.get_eigvals())
+
+    def test_decomposition_deprecation(self):
+        """Test that old-style staticmethod decompositions raise a warning"""
+        dev = qml.device("default.qubit", wires=1)
+
+        class MyOp(Operation):
+            num_wires = 1
+            num_params = 2
+
+            @staticmethod
+            def decomposition(*params, wires):
+                qml.RY(params[0], wires=wires[0])
+                qml.PauliZ(wires=wires[0])
+                qml.RX(params[1], wires=wires[0])
```

Comment:
I think we just need another case with no params here...

Reply:
oh oops, I didn't even spot this

Reply:
done!